get_hrc.py

```diff
@@ -139,44 +143,49 @@ def main():
     newdat, hrc_bad = format_proton_data(dat, descrs=descrs)
 
     # Use the 7-day file if there is a gap
-    if lasttime < newdat['time'][0]:
-        print("Warning: Data gap or error in GOES proton data.  Fetching 7-day JSON file")
+    if lasttime < newdat["time"][0]:
+        print(
+            "Warning: Data gap or error in GOES proton data.  Fetching 7-day JSON file"
+        )
         dat = get_json_data(URL_7D)
         newdat, hrc_bad = format_proton_data(dat, descrs=descrs)
 
-    with tables.open_file(args.h5, mode='a',
-                          filters=tables.Filters(complevel=5, complib='zlib')) as h5:
+    with tables.open_file(
+        args.h5, mode="a", filters=tables.Filters(complevel=5, complib="zlib")
+    ) as h5:
         try:
             table = h5.root.data
-            ok = newdat['time'] > lasttime
+            ok = newdat["time"] > lasttime
             h5.root.data.append(newdat[ok])
         except tables.NoSuchNodeError:
-            table = h5.create_table(h5.root, 'data', newdat,
-                                "HRC Antico shield + GOES", expectedrows=2e7)
+            table = h5.create_table(
+                h5.root, "data", newdat, "HRC Antico shield + GOES", expectedrows=2e7
+            )
         h5.root.data.flush()
 
     # Also write the mean of the last three values (15 minutes) to
     # hrc_shield.dat.  Only include good values.
-    times = DateTime(newdat['time'][-3:]).unix
-    hrc_shield = newdat['hrc_shield'][-3:]
+    times = DateTime(newdat["time"][-3:]).unix
+    hrc_shield = newdat["hrc_shield"][-3:]
     ok = ~hrc_bad[-3:]
     if len(hrc_shield[ok]) > 0:
         Path(args.data_dir).mkdir(exist_ok=True)
-        with open(Path(args.data_dir, 'hrc_shield.dat'), 'w') as f:
+        with open(Path(args.data_dir, "hrc_shield.dat"), "w") as f:
             print(hrc_shield[ok].mean(), times[ok].mean(), file=f)
 
     # For GOES earlier than 16:
     # for colname, scale, filename in zip(
     #     ('p2', 'p5'), (3.3, 12.0), ('p4gm.dat', 'p41gm.dat')):
     # GOES-16, ``scale`` TBD
     for colname, scale, filename in zip(
-        ('p4', 'p7'), (3.3, 12.0), ('p4gm.dat', 'p41gm.dat')):
+        ("p4", "p7"), (3.3, 12.0), ("p4gm.dat", "p41gm.dat"), strict=False
```

Comment:
It seems a little odd to be explicit about `strict=False`, but that matches the default behavior so that's fine.

Reply:
That's a ruff thing, trying to highlight that the default for `zip` is prone to subtle bugs. In this case it is a bit silly but harmless.

Reply:
Yeah, I knew this was a ruff thing in context - though I wondered if it would make sense to just use strict=True in these use cases if one is going to bother to add the strict keyword.